sherpa/stats/__init__.py

```diff
@@ -734,7 +735,17 @@ def calc_staterror(self, data):
         return self.errfunc(data)
 
     def calc_stat(self, data, model):
-        raise StatErr('nostat', self.name, 'calc_stat()')
+        data, model = self._validate_inputs(data, model)
+        fitdata = data.to_fit(staterrfunc=self.calc_staterror)
+        modeldata = data.eval_model_to_fit(model)
+        if not self._statfuncset:
```

Comment:
Yes, that's part of the refactoring I was mentioning, but thanks... it's been a while since I looked at this, there is no guarantee I would have remembered.